grr/server/grr_response_server/flows/general/osquery.py

```diff
@@ -19,17 +23,43 @@ class OsqueryFlow(flow_base.FlowBase):
   behaviours = flow_base.BEHAVIOUR_BASIC
 
   args_type = rdf_osquery.OsqueryArgs
+  progress_type = rdf_osquery.OsqueryProgress
 
   def Start(self):
     super(OsqueryFlow, self).Start()
+    self.state.progress = rdf_osquery.OsqueryProgress()
+
     self.CallClient(
         server_stubs.Osquery,
         request=self.args,
         next_state=compatibility.GetName(self.Process))
 
-  def Process(self, responses):
+  def Process(self, responses: flow_responses.Responses[rdf_osquery.OsqueryResult]) -> None:
     if not responses.success:
       raise flow_base.FlowError(responses.status)
 
+    self.UpdateProgress(responses)
+
     for response in responses:
       self.SendReply(response)
+
+  def GetProgress(self) -> rdf_osquery.OsqueryProgress:
+    return self.state.progress
+
+  def UpdateProgress(self, responses: flow_responses.Responses[rdf_osquery.OsqueryResult]) -> None:
```

Comment:
Throughout the codebase autoformat with 80 characters max line length. Please use this line length limit throughout the file (do you think you can configure your autoformatter with the same setting?)